firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClient.java

```diff
@@ -86,30 +94,42 @@
   private final long connectTimeoutInSeconds;
   private final long readTimeoutInSeconds;
 
-  /** Creates a client for {@link #fetch}ing data from the Firebase Remote Config server. */
+  private Transport<ClientLogEvent> transport;
+
+  /**
+   * Creates a client for {@link #fetch}ing data from the Firebase Remote Config server.
+   */
   public ConfigFetchHttpClient(
       Context context,
       String appId,
       String apiKey,
       String namespace,
       long connectTimeoutInSeconds,
-      long readTimeoutInSeconds) {
+      long readTimeoutInSeconds,
+      TransportFactory transportFactory) {
     this.context = context;
     this.appId = appId;
     this.apiKey = apiKey;
     this.projectNumber = extractProjectNumberFromAppId(appId);
     this.namespace = namespace;
     this.connectTimeoutInSeconds = connectTimeoutInSeconds;
     this.readTimeoutInSeconds = readTimeoutInSeconds;
+    this.transport =
```

Comment:
Since transport instantiation is orthogonal to fetch logic, can we inject a `Transport` instance, rather than injecting the factory?

Reply:
I've decided to inject a ConfigLogger class instead that uses the Transport instance.